src/components/game.js

```diff
@@ -28,25 +41,61 @@ const Row = styled.div`
   display: block;
 `;
 
+class Coordinates {
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+  }
+}
+
+// this class is used to store information about what happened to a tile in this move
+// It helps properly display animations of movement of tiles
+class MovementInfo {
+  constructor(fromCoords, toCoords, fromTileValue, toTileValue) {
+    this.fromCoords = fromCoords;
+    this.toCoords = toCoords;
+    this.fromTileValue = fromTileValue;
+    this.toTileValue = toTileValue;
+  }
+}
 
 class Game extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       board: this.initBoard(),
+      // this list stores information about every tile
+      // its initial and final coords and value
+      // eslint-disable-next-line react/no-unused-state
+      movedList: [],
+      // eslint-disable-next-line react/no-unused-state
+      isGameOver: false, // it is used. EsLint went nuts
     };
   }
 
   componentDidMount() {
     document.addEventListener('keydown', this.handleKeyDown, false);
-    this.newTile();
-    this.newTile();
+    // At the beginning of the game 2 tiles are needed.
+    // For them to load properly we use another newTile function in callback
+    this.placeNewTile(this.placeNewTile);
   }
 
   componentWillUnmount() {
     document.removeEventListener('keydown', this.handleKeyDown, false);
   }
 
+  newGame = () => {
+    this.setState({ board: this.initBoard() });
```

Comment:
jak chcesz dużo na raz ustawić to po prostu podajesz więcej rzeczy w obiekcie w setState, takie kolejkowanie nie ma sensu. Np:
`this.setState({prop1: 'lol'; prop2: 'eee'})`

Reply:
zrobione